src/main/scala/is/hail/expr/FunctionRegistry.scala

```diff
@@ -967,10 +967,9 @@ object FunctionRegistry {
     "stop" -> "Generate numbers up to, but not including this number.",
     "step" -> "Difference between each number in the sequence.")
 
-  registerCode("Call", (gt: Code[Int]) => CM.ret(gt),
-    """
-    Construct a :ref:`call` from an integer.
-    """, "gt" -> "integer")(int32Hr, callHr)
+  register("Call", { (gt: Int) => Call2.fromUnphasedDiploidIndex(gt) }, null)(int32Hr, callHr)
+  register("Call", { (alleles: IndexedSeq[Int], phased: Boolean) => CallN(alleles.toArray, phased) }, null)(arrayHr(int32Hr), boolHr, callHr)
+  register("Call0", { (phased: Boolean) => Call0(phased) }, null)(boolHr, callHr)
```

Comment:
I think I need more constructors here. It would be great if we could just pass the array of alleles, but I'm concerned about performance. I think what I need is for the python constructor to be varargs and then I call the correct constructor based on the length. Example:

```
c = functions.call(phased=True, ds.GT[0], ds.GT[1])
would call in function registry Call(i: Int, j: Int, phased: Boolean) => Call2(i, j, phased)
```

Reply:
I'm OK with the number suffix, but basically we should have Call0, Call1 and Call2 (or just Call overloaded).  And I agree the call constructor should be *args.